### PR TITLE
add configuration for course list, collection

### LINKED ISSUE #142: renaming of course collections
We want to both rename the things that are currently called 'course collections' and also introduce a new collection of these renamed entities.

- `course collections` -> `course lists`
- new entity: `course collections`, which consists of a list of course lists alongside a description and whatnot

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #142 

#### What's this PR do?

This PR adds configuration for a course collection gallery content type. This is basically a "collection of collections," and will render w/ a description and then the collections which have been added to it.

The idea is that this will be a good way to represent some of the more complicated collections that currently exist on production OCW, where there are different sections and whatnot. Basically, course authors would author each of those sections as a separate course collection, with a description and a list of courses, and then add them to a course collection gallery in the desired order.

#### How should this be manually tested?

You could paste this config into Studio and make sure it works properly.

### CODE REVIEW

File: ocw-www/ocw-studio.yaml [hunk 1]
```diff
@@ -76,6 +76,37 @@ collections:
         name: courses
         widget: website-collection
 
+  - category: Content
+    folder: content/collectionsgallery
+    label: "Course Collection Galleries"
+    name: collections-gallery
```
Comment (Ferdi) [inline ocw-www/ocw-studio.yaml:82]: course-collection-gallery ?

**Review by abeglova — APPROVED** [review]
LGTM

File: ocw-www/ocw-studio.yaml [hunk 2]
```diff
@@ -76,6 +76,37 @@ collections:
         name: courses
         widget: website-collection
 
+  - category: Content
+    folder: content/collectiongalleries
```
Comment (Ferdi) [inline ocw-www/ocw-studio.yaml:80]: Not sure about the "collectiongalleries" in the url .

Comment (alicewriteswrongs) [reply ocw-www/ocw-studio.yaml:80]: this brings up a larger question of what to call them. Is 'Course Gallery' the term we want? I've tried to think of a few alternatives that include the connotation of order and structure within (i.e. this is not just a big list of courses, but rather a list-of-lists with some additional structure in the form of featured courses, a description, etc).

What do you think of calling it something like a 'Course Compendium'? I was trying to think of something that has a bit of a 'larger' connotation than just 'Collection'. If we went with that they could be under `/compendiums` for the URL. 

We could also go in another direction that gets more at the _function_ of these things. It seems we're mostly dealing with a large, structured guide to a particular topic or area. So maybe these should be called a 'Topic Guide', 'Subject Guide', or something that captures a little bit more of the intent for how this should be used.

Anyway, I'm not sure, but I think that 'Course Gallery' may not be a totally adequate name.

Comment (Ferdi) [reply ocw-www/ocw-studio.yaml:80]: @dseaton ?

**Comment by alicewriteswrongs** [comment]
@dseaton @Ferdi I just updated this to reflect our recent conversations about 'course lists' and 'collections'.

@abeglova accordingly, I think it might be good if you could take another look 🙂

**Pull request merged**